### ISSUE TITLE
GET errors due to Connection reset under load

### ISSUE THREAD

**Issue by derjust** [GoogleCloudPlatform/cloud-bigtable-client#482]
While running a large test with a cluster of 60, we see those exceptions bubbling up to our user code

```
 2015-09-14 16:49:12,018 WARN  | [pool-1-thread-8] (c.s.a.b.LifecycleBuilder:139) | An IOException has occurred Failed to perform operation. Operation='get', projectId='demo', tableName='someTable', rowKey='adbb735f-4718-45d9-badf-b106a61f8ed0|UTXQ'
 java.io.IOException: Failed to perform operation. Operation='get', projectId='sungard-cat-demo', tableName='someTable', rowKey='adbb735f-4718-45d9-badf-b106a61f8ed0|UTXQ'
 at com.google.cloud.bigtable.hbase.BigtableTable.get(BigtableTable.java:199) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.sungard.advtech.bigtable_df.LifecycleBuilder.processElement(LifecycleBuilder.java:95) ~[classes-KrHrvliWr0XWi3H_iWOlEg.zip:na]
 at com.google.cloud.dataflow.sdk.util.DoFnRunner.invokeProcessElement(DoFnRunner.java:189) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.util.DoFnRunner.processElement(DoFnRunner.java:171) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.runners.worker.ParDoFnBase.processElement(ParDoFnBase.java:193) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.util.common.worker.ParDoOperation.process(ParDoOperation.java:52) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.util.common.worker.OutputReceiver.process(OutputReceiver.java:52) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.util.common.worker.ReadOperation.runReadLoop(ReadOperation.java:171) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.util.common.worker.ReadOperation.start(ReadOperation.java:117) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.util.common.worker.MapTaskExecutor.execute(MapTaskExecutor.java:66) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.runners.worker.DataflowWorker.executeWork(DataflowWorker.java:234) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.runners.worker.DataflowWorker.doWork(DataflowWorker.java:171) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.runners.worker.DataflowWorker.getAndPerformWork(DataflowWorker.java:137) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.runners.worker.DataflowWorkerHarness$WorkerThread.call(DataflowWorkerHarness.java:147) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at com.google.cloud.dataflow.sdk.runners.worker.DataflowWorkerHarness$WorkerThread.call(DataflowWorkerHarness.java:132) [google-cloud-dataflow-java-sdk-all-1.0.0-rHz39Me5Bgx6ma4iSO7HHQ.jar:na]
 at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_60-ea]
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_60-ea]
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_60-ea]
 at java.lang.Thread.run(Thread.java:745) [na:1.8.0_60-ea]
 Caused by: com.google.cloud.bigtable.grpc.io.IOExceptionWithStatus: Error in response stream
 at com.google.cloud.bigtable.grpc.scanner.ResultQueueEntry.getResponseOrThrow(ResultQueueEntry.java:66) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.cloud.bigtable.grpc.scanner.StreamingBigtableResultScanner$ResponseQueueReader.getNextMergedRow(StreamingBigtableResultScanner.java:75) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.cloud.bigtable.grpc.scanner.StreamingBigtableResultScanner.next(StreamingBigtableResultScanner.java:136) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.cloud.bigtable.grpc.scanner.StreamingBigtableResultScanner.next(StreamingBigtableResultScanner.java:32) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.cloud.bigtable.grpc.scanner.ResumingStreamingResultScanner.next(ResumingStreamingResultScanner.java:90) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.cloud.bigtable.grpc.scanner.ResumingStreamingResultScanner.next(ResumingStreamingResultScanner.java:43) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.cloud.bigtable.hbase.BigtableTable.get(BigtableTable.java:195) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 ... 18 common frames omitted
 Caused by: io.grpc.StatusRuntimeException: UNKNOWN
 at io.grpc.Status.asRuntimeException(Status.java:428) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:264) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at io.grpc.ClientCallImpl$ClientStreamListenerImpl$3.run(ClientCallImpl.java:293) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at io.grpc.internal.SerializingExecutor$TaskRunner.run(SerializingExecutor.java:154) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 ... 3 common frames omitted
 Caused by: java.io.IOException: Connection reset by peer
 at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_60-ea]
 at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_60-ea]
 at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_60-ea]
 at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_60-ea]
 at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_60-ea]
 at com.google.bigtable.repackaged.io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:311) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.bigtable.repackaged.io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:854) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.bigtable.repackaged.io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:242) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.bigtable.repackaged.io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:115) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.bigtable.repackaged.io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:510) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.bigtable.repackaged.io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:467) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.bigtable.repackaged.io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:381) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.bigtable.repackaged.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 at com.google.bigtable.repackaged.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:703) ~[bigtable-hbase-1.0-0.2.2-SJ-UlknvKCKvgD59j6kCsTcCg.jar:na]
 ... 1 common frames omitted
```

Just as a note:
`bigtable-hbase-1.0-0.2.2-SJ.jar` is master + #480 + #481

**Comment by kevinsi4508** [comment]
https://github.com/grpc/grpc-java/issues/1053

**Comment by sduskis** [comment]
Our retries that we released with the 0.2.4 release should sufficiently fix this problem
